### DISCUSSION TITLE
pypowerwall no longer listening on 8675

### DISCUSSION THREAD

**Discussion by ViktorJp** [jasonacox/Powerwall-Dashboard#464] [Q&A] [answered]
For the life of me, something stopped working today right at noon for some crazy reason... I've rebooted my server, tried to restart the containers, tried an upgrade (which went successfully), but grafana is no longer showing current data, and the powerwall flow graph is no longer loading.  When I run 'verify.sh', I'm seeing this... the TZ is set across the board for "America/New_York".  I don't understand what else could be hanging it up?  Any ideas how else to troubleshoot this?

`Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): ERROR: Not Listening
 - Version: Unknown
 - Powerwall State: ERROR: Not Connected - Firmware Version: Unknown
 - Cloud Mode: NO
 - ERROR: Your pypowerwall settings are missing TZ.

Checking telegraf
----------------------------------------------------------------------------
 - Config File telegraf.conf: GOOD
 - Local Config File telegraf.local: GOOD
 - Container (telegraf): GOOD
 - Version: Telegraf 1.28.2 (git: HEAD@8d9cf395)

Checking influxdb
----------------------------------------------------------------------------
 - Config File influxdb.conf: GOOD
 - Environment File influxdb.env: GOOD
 - Container (influxdb): GOOD
 - Service (port 8086): GOOD
 - Filesystem (./influxdb): GOOD
 - Version: InfluxDB shell version: 1.8.10

Checking grafana
----------------------------------------------------------------------------
 - Config File grafana.env: GOOD
 - Container (grafana): GOOD
 - Service (port 9000): GOOD
 - Filesystem (./grafana): GOOD
 - Version: Grafana CLI version 9.1.2

Checking weather411
----------------------------------------------------------------------------
 - Skipped: weather411 not set up (missing weather/weather411.conf)

One or more tests failed.
`

**Comment by jasonacox** [comment, marked as answer]
Hi @ViktorJp 

The error means the `pypowerwall.env` is missing the `TZ=` setting. That file is created during `setup.sh` and should look like:

```
PW_EMAIL=email@example.com
PW_PASSWORD=password
PW_HOST=10.0.1.2
PW_TIMEZONE=America/Los_Angeles
TZ=America/Los_Angeles
PW_DEBUG=no
PW_STYLE=grafana-dark
```

The `TZ` line is relatively new but the `upgrade.sh` script should have added it if it was missing. It is odd that `verify.sh` is reporting that it is missing. It would be good to check that file to ensure it isn't corrupt.  It would also be good to see what pypowerwall says:

```bash
docker logs pypowerwall`
```

That should help explain why the container is not starting.  You can also run `setup.sh` to have it reset the settings.

**Comment by ViktorJp** [comment]
Thanks for your help, Jason!  That may be my problem... in my pypowerwall.env file, everything starts with PW_... including the PW_TIMEZONE... I noticed yours simply is referenced as "TZ="

```
PW_EMAIL=email@whatever.com
PW_PASSWORD=PWD
PW_HOST=192.168.86.63
PW_TIMEZONE=America/New_York
PW_DEBUG=no
PW_STYLE=grafana-dark
```
Would you recommend just manually adding this item TZ=?

So just some background... I'm running ubuntu as a vm on a windows server with hyper-v.  From day 1, I have always had major issues with needing to run certain commands with sudo, other times they would work without needing sudo... but it's sometimes real tough to get through an upgrade without it bombing out because of permission issues. Your scripts will exit 1 when sudo is detected, and have to be able to get around that at times.  I just don't know enough about Linux in general to get around this issue... one would think that if I install this as a sudo admin that things would just work.  I've followed your advice and steps on making sure the current user is a container admin, but in the end, at times have to step through your upgrade/setup scripts line by line, and run commands using sudo to make it through.  Not sure what else to try other than perhaps starting a new fresh ubuntu server all over... but needless to say, if the upgrade script was supposed to replace "PW_TIMEZONE=" with "TZ=", then this may have been a cause.

**Reply by ViktorJp** [reply]
Adding that TZ= seems to have done the trick after a docker restart!  Thanks for pointing this out, Jason!

```
Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): GOOD
 - Version: 0.7.7 Proxy t40
 - Powerwall State: CONNECTED - Firmware Version: 24.4.0
 - Cloud Mode: NO
```

**Comment by SCHibbard** [comment]
FYI I had a similar experience yesterday with weather411.  At 9:50 Central Time all weather data stopped., however direct calls to api.openweathermap.org worked fine.  Powerwall data was doing fine, as were data streams I have added for my provider's hourly energy pricing and data from my microinverters.  Before trouble shooting further, thought I should upgrade my sandbox install.  Went from 4.0.3 to 4.2.1 and after that all was well with the world so replicated on production install.  Assume it was the TZ variable ... ignorance worked in my favor this time.

**Comment by DMBlakeley** [comment]
Today, I experienced the Docker Desktop for Mac issue where Docker Desktop would not starting with and inaccurate malware warnings. 

I installed Docker Desktop 4.7.2 which was supposed to address the issue. Ever since installing, I cannot get the `pypowerwall` container to keep running (it is in a continuous restart loop). Troubleshooting:

- I have checked all the TZ settings as described above and still no luck. 
- Have uninstalled and reinstalled Docker Desktop followed by running `./setup.sh`. 
- I can ping the Powerwall at 192.168.91.1. 
- I have a Powerwall+ with Powerwall 2. Firmware version is 24.44.0.

I may or may not have corrupted the database but may be able to recover from backups (I have 2 ½ year of data).

I am at a loss and looking for some possible direction. Thanks!

**Additional Information**
----

**./verify.sh output**
---

Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): ERROR: Not Listening
 - Version: Unknown
 - Powerwall State: ERROR: Not Connected - Firmware: Unknown
 - Gateway TEDAPI: Available (192.168.91.1)
 - TEDAPI Vitals: Not Connected 
 - Cloud Mode: NO

**pypowerall logs (following repeats):**
---
2025-01-16 22:32:18 During handling of the above exception, another exception occurred:
2025-01-16 22:32:18 
2025-01-16 22:32:18 Traceback (most recent call last):
2025-01-16 22:32:18   File "/app/server.py", line 210, in <module>
2025-01-16 22:32:18     site_name = pw.site_name() or "Unknown"
2025-01-16 22:32:18   File "/usr/local/lib/python3.10/site-packages/pypowerwall/__init__.py", line 498, in site_name
2025-01-16 22:32:18     payload = self.poll('/api/site_info/site_name')
2025-01-16 22:32:18   File "/usr/local/lib/python3.10/site-packages/pypowerwall/__init__.py", line 318, in poll
2025-01-16 22:32:18     payload = self.client.poll(api, force, recursive, raw)
2025-01-16 22:32:18   File "/usr/local/lib/python3.10/site-packages/pypowerwall/local/pypowerwall_local.py", line 196, in poll
2025-01-16 22:32:18     self._get_session()
2025-01-16 22:32:18   File "/usr/local/lib/python3.10/site-packages/pypowerwall/local/pypowerwall_local.py", line 114, in _get_session
2025-01-16 22:32:18     raise LoginError("Invalid Powerwall Login")
2025-01-16 22:32:18 pypowerwall.local.exceptions.LoginError: Invalid Powerwall Login

**Reply by jasonacox** [reply]
It is not able to log in to the Powerwall portal. Go to 192.168.91.1 in your browser to see if you can log in using the email and password in. your  pypowerwall.env file. You may have to dismiss the upgrade notices on the Powerwall portal page.

```bash
# Powerwall Settings
PW_EMAIL=
PW_PASSWORD=
PW_HOST=192.168.91.1
PW_TIMEZONE=America/Los_Angeles
TZ=America/Los_Angeles

# Powerwall WiFi Gateway Password for TEDAPI Access Mode
PW_GW_PWD=XXXXXXXX
```

Another option, you can clear out the PW_EMAIL and PW_PASSWORD fields an the connection will only use TEDAPI (PW_GW_PWD) to get stats. it switches it to "full" mode instead of "hybrid" which means there are a few data points that are computed instead of pulled and you will be missing some of the "current" data.

**Comment by DMBlakeley** [comment]
Thanks for the response! 

I was able to login at 192.168.91.1 using my email and the last 5 characters of the PW_GW_PW after dismissing the upgrade notice.

In the Powerwall Settings, I had the Tesla account password as the PW_PASSWORD. It has been this way for quite some time without issue! Changed to the 5 character password, ran `./setup.sh` and was back in business after many hours of getting no where!

 All good again!

**Reply by jasonacox** [reply]
Thanks @DMBlakeley ! I seem to recall others seeing this. I do wonder if it is the result of a firmware upgrade.
